helix-term/src/commands/typed.rs

```diff
@@ -5,14 +5,20 @@ use crate::job::Job;
 
 use super::*;
 
+use globset::{Glob, GlobBuilder, GlobSet};
 use helix_core::fuzzy::fuzzy_match;
 use helix_core::indent::MAX_INDENT;
 use helix_core::{encoding, line_ending, shellwords::Shellwords};
+use helix_stdx::env::current_working_dir;
 use helix_view::document::DEFAULT_LANGUAGE_NAME;
 use helix_view::editor::{Action, CloseError, ConfigEvent};
 use serde_json::Value;
 use ui::completers::{self, Completer};
 
+// The maximum number of files to open with globbing to avoid freezing while trying to open too many files.
+// It is kind of arbitrary and can be raised or offered as a configuration option.
+const GLOBBING_MAX_N_FILES: usize = 32;
```

Comment:
I think this limit should be higher. Maybe 64. I doubt that would freeze to ui much and I had some niche cases where I really did want to open that many files. Wouldn't want to add a confog for this